### ISSUE TITLE
CriterionDataParallel example

### ISSUE THREAD

**Issue by iliadsouti** [zhanghang1989/PyTorch-Encoding#19]
Can you give an example of how to use CriterionDataParallel?
Thanks

**Comment by zhanghang1989** [comment]
Use together with modelParallel or selfParallel.

**Comment by iliadsouti** [comment]
I do not understand. Can you make an example similar with the one in modelParallel : 
net = encoding.nn.ModelDataParallel(model, device_ids=[0, 1, 2])
?

**Comment by cbelth** [comment]
So something like: 
`net = encoding.nn.CriterionDataParallel(encoding.nn.ModelDataParallel(model, device_ids=[0, 1, 2]))`?

**Comment by zhanghang1989** [comment]
model = ModelDataParallel(model)
criterion = CriterionDataParallel(criterion)

y = model(x)
loss = criterion(y, target)

**Comment by facetohard** [comment]
@zhanghang1989 I test the example code you wrote. The type of output of `model(x)` is `list()` and its length is the num of device. Does it right.

**Comment by wangxiaodong1021** [comment]
Also meet the same error.

**Comment by zhanghang1989** [comment]
Replace https://github.com/zhanghang1989/PyTorch-Encoding/blob/master/encoding/parallel.py#L203
with ``return Reduce()(*outputs)``. This will be updated in next version.

**Comment by zhanghang1989** [comment]
This issue has been resolved in favor of releasing PyTorch compatible Synchronized Cross-GPU [encoding.nn.BatchNorm2d](http://hangzh.com/PyTorch-Encoding/syncbn.html) and the [example](https://github.com/zhanghang1989/PyTorch-SyncBatchNorm).
